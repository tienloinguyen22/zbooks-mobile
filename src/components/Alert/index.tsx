import React from 'react';
import RootSiblings from 'react-native-root-siblings';
import { AlertContainer, AlertAction, AlertType } from '../AlertContainer';

interface ShowParams {
  type: AlertType;
  title?: string;
  message: string;
  actions?: AlertAction[];
  onPressClose?: () => void;
  closeable?: boolean;
}

let alert: RootSiblings | undefined;
const hide = (): void => {
  alert && alert.destroy();
};

const show = (props: ShowParams): RootSiblings => {
  alert && hide();
  alert = new RootSiblings(<AlertContainer {...props} onPressClose={props.onPressClose || hide} />);
  return alert;
};

export const Alert = {
  show,
  hide,
};
