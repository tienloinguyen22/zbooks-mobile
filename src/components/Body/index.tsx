import { Body } from 'native-base';

export { Body };
