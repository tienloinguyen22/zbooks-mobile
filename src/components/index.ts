export * from './Body';
export * from './Button';
export * from './Card';
export * from './CardItem';
export * from './Container';
export * from './Content';
export * from './DatePicker';
export * from './ErrorBoundary';
export * from './ErrorText';
export * from './Field';
export * from './Footer';
export * from './Header';
export * from './Icon';
export * from './Image';
export * from './Input';
export * from './Item';
export * from './Label';
export * from './Left';
export * from './List';
export * from './ListItem';
export * from './Loading';
export * from './Picker';
export * from './Right';
export * from './ScrollView';
export * from './Text';
export * from './Touchable';
export * from './Thumbnail';
export * from './Tooltip';
export * from './View';
export * from './WithStore';
export * from './WithScreen';
export * from './Alert';
export * from './Animation';
export * from './AlertContainer';
