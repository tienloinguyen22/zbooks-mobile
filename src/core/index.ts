export * from './i18n';
export * from './themes/colors';
export * from './interfaces/ScreenProps';
export * from './interfaces/LoginType';
export * from './interfaces/User';
export * from './interfaces/Resource';
export * from './interfaces/Theme';
export * from './helpers/show_notification';
export * from './helpers/sleep';
export * from './helpers/record_error';
export * from './helpers/configure_google_sign_in';
export * from './helpers/combine_styles';
export * from './helpers/configure_google_sign_in';
export * from './helpers/handle_error';
export * from './helpers/get_primary_color';
export * from './helpers/format_date';
export * from './screen_names';
