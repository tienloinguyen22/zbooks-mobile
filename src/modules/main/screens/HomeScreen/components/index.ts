export * from './NavigationSample';
export * from './IconSample';
export * from './RematchSample';
export * from './NotificationSample';
export * from './CrashSample';
export * from './PickerSample';
export * from './AnalyticsSample';
export * from './TextStyleSample';
