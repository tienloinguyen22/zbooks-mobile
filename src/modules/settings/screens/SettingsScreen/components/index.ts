export * from './SettingMenuItem';
export * from './LogoutModal';
