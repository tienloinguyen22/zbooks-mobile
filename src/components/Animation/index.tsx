import React from 'react';
import { Resource } from '@app/core';
import LottieView from 'lottie-react-native';

interface Props {
  width: number;
  height: number;
  source: Resource;
  autoPlay?: boolean;
  loop?: boolean;
}

export const Animation = (props: Props): JSX.Element => {
  return (
    <LottieView
      source={props.source}
      style={{
        width: props.width,
        height: props.height,
      }}
      autoPlay={props.autoPlay === true}
      loop={props.loop === true}
    />
  );
};
