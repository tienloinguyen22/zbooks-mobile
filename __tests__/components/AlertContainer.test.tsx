import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { AlertContainer, View, Button, Text } from '@app/components';
import { THEME_DARK, THEME_LIGHT } from '@app/core';
import { mockTheme } from '../test_helpers';

const mockModal = (modalProps: { onBackdropPress: () => void; children: React.ReactNode }): JSX.Element => {
  const onPress = jest.fn().mockImplementation(() => {
    modalProps.onBackdropPress();
  });
  return (
    <View>
      {modalProps.children}
      <Button onPress={onPress}>
        <Text>Close</Text>
      </Button>
    </View>
  );
};
jest.mock('react-native-modal', () => {
  const modal = ({ onBackdropPress, children }: { onBackdropPress: () => void; children: React.ReactNode }): Element =>
    mockModal({
      onBackdropPress,
      children,
    });
  return modal;
});

describe('components/AlertContainer', () => {
  const alertTitle = 'Title';
  const message = 'Message';
  const closeText = 'dialog.close';
  const customButtonText = 'customButton';
  const cancelText = 'cancel';
  const closeAction = jest.fn();
  const cancelAction = jest.fn();
  const customAction = jest.fn();
  const props = {
    title: alertTitle,
    message,
    visible: true,
    onPressClose: closeAction,
  };
  const propWithNoTitle = {
    message,
    visible: true,
    onPressClose: closeAction,
  };
  const propWithActions = {
    title: alertTitle,
    message,
    visible: true,
    onPressClose: closeAction,
    actions: [
      {
        title: customButtonText,
        onPress: customAction,
        special: true,
      },
      {
        title: cancelText,
        onPress: cancelAction,
        special: true,
      },
    ],
  };

  beforeEach(() => {
    mockTheme(THEME_DARK);
  });

  it('renders successfully', async () => {
    const { baseElement, getByText } = render(<AlertContainer {...props} type='INFO' />);

    expect(getByText(alertTitle)).toBeDefined();
    expect(getByText(message)).toBeDefined();
    expect(getByText(closeText)).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });

  it('renders successfully with light theme', async () => {
    mockTheme(THEME_LIGHT);
    const { baseElement, getByText } = render(<AlertContainer {...props} type='INFO' />);

    expect(getByText(alertTitle)).toBeDefined();
    expect(getByText(message)).toBeDefined();
    expect(getByText(closeText)).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });

  it('renders info alert successfully', async () => {
    const { baseElement } = render(<AlertContainer {...propWithNoTitle} type='INFO' />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders warning alert successfully', async () => {
    const { baseElement } = render(<AlertContainer {...propWithNoTitle} type='WARNING' />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders error alert successfully', async () => {
    const { baseElement } = render(<AlertContainer {...propWithNoTitle} type='ERROR' />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders success alert successfully', async () => {
    const { baseElement } = render(<AlertContainer {...propWithNoTitle} type='SUCCESS' />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders successfully with its custom actions', async () => {
    const { baseElement, getByText } = render(<AlertContainer {...propWithActions} type='INFO' />);
    expect(getByText(customButtonText)).toBeDefined();
    expect(getByText(cancelText)).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });

  it('handles press when click default close button', async () => {
    const { getByText } = render(<AlertContainer {...props} type='INFO' />);
    const closeButton = getByText(closeText);
    fireEvent.press(closeButton);

    expect(props.onPressClose).toBeCalledTimes(1);
  });

  it('handles press when clicking custom button', async () => {
    const { getByText } = render(<AlertContainer {...propWithActions} type='INFO' />);
    const customButton = getByText(customButtonText);
    fireEvent.press(customButton);

    expect(customAction).toBeCalledTimes(1);
  });

  it('hide alert when clicking outside modal', async () => {
    const { getByText } = render(<AlertContainer {...props} type='INFO' />);
    // use a fake button to trigger event
    const button = getByText('Close');
    fireEvent.press(button);

    expect(props.onPressClose).toBeCalledTimes(1);
  });
});
