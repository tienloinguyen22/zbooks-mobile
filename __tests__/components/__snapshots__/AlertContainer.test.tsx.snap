// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AlertContainer renders error alert successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 1334,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 750,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 5,
              },
              Object {
                "backgroundColor": "#1A2237",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FC4173",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 5,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingVertical": 10,
                  "textAlign": "center",
                  "textTransform": "none",
                }
              }
            >
              dialog.error
            </Text>
            <View
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textTransform": "none",
                },
                Object {
                  "paddingVertical": 15,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Message
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginBottom": 10,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "black",
                    "borderRadius": 20,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 80,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#fff",
                      "fontWeight": "bold",
                      "textTransform": "none",
                    }
                  }
                  white={true}
                >
                  dialog.close
                </Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderRadius": 20,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "width": 200,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "textTransform": "none",
                }
              }
            >
              Close
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;

exports[`components/AlertContainer renders info alert successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 1334,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 750,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 5,
              },
              Object {
                "backgroundColor": "#1A2237",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 5,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingVertical": 10,
                  "textAlign": "center",
                  "textTransform": "none",
                }
              }
            >
              dialog.info
            </Text>
            <View
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textTransform": "none",
                },
                Object {
                  "paddingVertical": 15,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Message
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginBottom": 10,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "black",
                    "borderRadius": 20,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 80,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#fff",
                      "fontWeight": "bold",
                      "textTransform": "none",
                    }
                  }
                  white={true}
                >
                  dialog.close
                </Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderRadius": 20,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "width": 200,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "textTransform": "none",
                }
              }
            >
              Close
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;

exports[`components/AlertContainer renders success alert successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 1334,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 750,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 5,
              },
              Object {
                "backgroundColor": "#1A2237",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#24DF98",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 5,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingVertical": 10,
                  "textAlign": "center",
                  "textTransform": "none",
                }
              }
            >
              dialog.success
            </Text>
            <View
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textTransform": "none",
                },
                Object {
                  "paddingVertical": 15,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Message
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginBottom": 10,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "black",
                    "borderRadius": 20,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 80,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#fff",
                      "fontWeight": "bold",
                      "textTransform": "none",
                    }
                  }
                  white={true}
                >
                  dialog.close
                </Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderRadius": 20,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "width": 200,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "textTransform": "none",
                }
              }
            >
              Close
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;

exports[`components/AlertContainer renders successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 1334,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 750,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 5,
              },
              Object {
                "backgroundColor": "#1A2237",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 5,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingVertical": 10,
                  "textAlign": "center",
                  "textTransform": "none",
                }
              }
            >
              Title
            </Text>
            <View
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textTransform": "none",
                },
                Object {
                  "paddingVertical": 15,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Message
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginBottom": 10,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "black",
                    "borderRadius": 20,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 80,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#fff",
                      "fontWeight": "bold",
                      "textTransform": "none",
                    }
                  }
                  white={true}
                >
                  dialog.close
                </Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderRadius": 20,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "width": 200,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "textTransform": "none",
                }
              }
            >
              Close
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;

exports[`components/AlertContainer renders successfully with its custom actions 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 1334,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 750,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 5,
              },
              Object {
                "backgroundColor": "#1A2237",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 5,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingVertical": 10,
                  "textAlign": "center",
                  "textTransform": "none",
                }
              }
            >
              Title
            </Text>
            <View
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textTransform": "none",
                },
                Object {
                  "paddingVertical": 15,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Message
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 10,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "black",
                    "borderRadius": 20,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 80,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#fff",
                      "fontWeight": "bold",
                      "textTransform": "none",
                    }
                  }
                  white={true}
                >
                  customButton
                </Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "black",
                    "borderRadius": 20,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 80,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#fff",
                      "fontWeight": "bold",
                      "textTransform": "none",
                    }
                  }
                  white={true}
                >
                  cancel
                </Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderRadius": 20,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "width": 200,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "textTransform": "none",
                }
              }
            >
              Close
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;

exports[`components/AlertContainer renders successfully with light theme 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 1334,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 750,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 5,
              },
              Object {
                "backgroundColor": "#fff",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 5,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingVertical": 10,
                  "textAlign": "center",
                  "textTransform": "none",
                }
              }
            >
              Title
            </Text>
            <View
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textTransform": "none",
                },
                Object {
                  "paddingVertical": 15,
                  "textAlign": "center",
                },
                Object {
                  "color": "#222C44",
                },
              ]
            }
          >
            Message
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginBottom": 10,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "black",
                    "borderRadius": 20,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 80,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#fff",
                      "fontWeight": "bold",
                      "textTransform": "none",
                    }
                  }
                  white={true}
                >
                  dialog.close
                </Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderRadius": 20,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "width": 200,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "textTransform": "none",
                }
              }
            >
              Close
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;

exports[`components/AlertContainer renders warning alert successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 1334,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 750,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 5,
              },
              Object {
                "backgroundColor": "#1A2237",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FDA929",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 5,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingVertical": 10,
                  "textAlign": "center",
                  "textTransform": "none",
                }
              }
            >
              dialog.warning
            </Text>
            <View
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textTransform": "none",
                },
                Object {
                  "paddingVertical": 15,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Message
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginBottom": 10,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "black",
                    "borderRadius": 20,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 80,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#fff",
                      "fontWeight": "bold",
                      "textTransform": "none",
                    }
                  }
                  white={true}
                >
                  dialog.close
                </Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderRadius": 20,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "width": 200,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "textTransform": "none",
                }
              }
            >
              Close
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;
